caustic/lenses/tnfw.py

```diff
@@ -0,0 +1,303 @@
+
+    @unpack(2)
+    def projected_mass(
+            self, r: Tensor, z_s: Tensor, z_l, x0, y0, m, c, t, *args, params: Optional["Packed"] = None, **kwargs
+    ) -> Tensor:
+        """
+        Total projected mass (Msol) within a radius r (Mpc).
+
+        Args:
+            z_l (Tensor): Redshift of the lens.
+            x0 (Tensor): Center of lens position on x-axis (arcsec). 
+            y0 (Tensor): Center of lens position on y-axis (arcsec). 
+            m (Tensor): Mass of the lens (Msol).
+            c (Tensor): Concentration parameter of the lens (r200/rs for a classic NFW).
+            t (Tensor): Truncation scale. Ratio of truncation radius to scale radius (rt/rs).
+            params (dict): Dynamic parameter container.
+
+        Returns:
+            Tensor: Integrated mass projected in infinite cylinder within radius r.
+        """
+        rs = self.get_scale_radius(params)
+        x = r / rs
+        F = self._F(x)
+        L = self._L(x, t)
+        a1 = t**2 / (t**2 + 1)**2
+        a2 = (t**2 + 1 + 2*(x**2 - 1)) * F
+        a3 = t * torch.pi
+        a4 = (t**2 - 1) * t.log()
+        a5 = (t**2 + x**2).sqrt() * (-torch.pi + (t**2 - 1) * L / t)
+        S = self.get_M0(params)
+        return S * a1 * (a2 + a3 + a4 + a5)
+        
+    @unpack(3)
+    def physical_deflection_angle(
+            self, x: Tensor, y: Tensor, z_s: Tensor, z_l, x0, y0, m, c, t, *args, params: Optional["Packed"] = None, **kwargs
+    ) -> tuple[Tensor, Tensor]:
+        """Compute the physical deflection angle (arcsec) for this lens at
+        the requested position. Note that the NFW profile is more
+        naturally represented as a physical deflection angle, this is
+        easily internally converted to a reduced deflection angle.
+
+        Args:
+            z_l (Tensor): Redshift of the lens.
+            x0 (Tensor): Center of lens position on x-axis (arcsec). 
+            y0 (Tensor): Center of lens position on y-axis (arcsec). 
+            m (Tensor): Mass of the lens (Msol).
+            c (Tensor): Concentration parameter of the lens (r200/rs for a classic NFW).
+            t (Tensor): Truncation scale. Ratio of truncation radius to scale radius (rt/rs).
+            params (dict): Dynamic parameter container.
+
+        Returns:
+            tuple[Tensor, Tensor]: The physical deflection angles in the x and y directions (arcsec).
+
+        """
+        d_l = self.cosmology.angular_diameter_distance(z_l, params)
+        rs = self.get_scale_radius(params)
+        x, y = translate_rotate(x, y, x0, y0)
+        r = ((x**2 + y**2).sqrt() + self.s) * d_l * arcsec_to_rad
+        theta = torch.arctan2(y,x)
+
+        # The below actually equally comes from eq 2.13 in Meneghetti notes
+        dr = self.projected_mass(r, z_s, params) / r # note dpsi(u)/du = 2x*dpsi(x)/dx when u = x^2
+        S = 4 * G_over_c2 * rad_to_arcsec
+        return S * dr * theta.cos(), S * dr * theta.sin()
```

Comment:
TNFW (and NFW actually) are just more naturally expressed in physical deflection angle. Converting to reduced deflection angle just involves a ratio of distances which is what happens automatically internally anyway. Users can still use the `reduced_deflection_angle` function.

As for the the 1/d_l I don't think that's needed. If you look at eq 2.13 from meneghetti it looks right. Also a 1/d_l already comes in since `r = theta * d_l * arcsec_to_rad`. and the expression has `1/r`